### PR TITLE
Add subject model

### PR DESCRIPTION
**Pull request by DenisZackharov**
### Summary
Add subject model and many-to-many through company_subjects model

### How it works

None.

### Test plan

None.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have added tests for new functionality
- [x] My pull request is shorter than 300 lines

### Deploy notes

Notes regarding deployment the contained body of work.
These should note any db migrations, ENV variables, services, scripts, etc.

### CODE REVIEW

File: app/models/subject.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    "Math",
+    "History",
+    "English Language",
+    "Russian Language",
+    "Physics",
+    "Chemistry",
+    "Biology",
+    "Informatics",
+    "Geography",
+    "Litrature",
```
Comment (RasimKhusaenov) [inline app/models/subject.rb:14]:
```suggestion
    "Literature",
```

File: app/models/subject.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    "Social Science",
+    "Physical Education"
+  ].freeze
+
+  has_many :company_subjects, dependent: :destroy
+  has_many :companies, through: :company_subjects
+
+  validates :name, presence: true, uniqueness: true
+
+  enumerize :name, in: SUBJECT_NAMES
```
Comment (RasimKhusaenov) [inline app/models/subject.rb:24]: Why are subject names limited to this array? I think there can be many different ones at the university at least

File: db/seeds/developement/subjects.seeds.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+Subject.find_or_create_by(name: "Math")
+Subject.find_or_create_by(name: "History")
+Subject.find_or_create_by(name: "English Language")
+Subject.find_or_create_by(name: "Russian Language")
+Subject.find_or_create_by(name: "Physics")
+Subject.find_or_create_by(name: "Chemistry")
+Subject.find_or_create_by(name: "Biology")
+Subject.find_or_create_by(name: "Informatics")
+Subject.find_or_create_by(name: "Geography")
+Subject.find_or_create_by(name: "Litrature")
```
Comment (RasimKhusaenov) [inline db/seeds/developement/subjects.seeds.rb:12]: Literature

**Review by RasimKhusaenov — APPROVED** [review]

**Review by egorpavlovs — APPROVED** [review]
Looks good!
NBD: I think here (or at the next PR) we can add tests for checking factories (`expect(subject).to be_valid`) and relations between models

**Pull request merged**